doc/authentication-methods/ldap.md

```diff
@@ -0,0 +1,149 @@
+* **Default:** `[{attr = "my_uid"}]`
+* **Example:** `uids = [{attr = "my_uid", format = "%u@example.org"}, {attr = "another_uid"}]`
+
+List of LDAP attributes that contain the user name (user's part of the JID), used to search for user accounts.
+They are used as alternatives - it is enough if one of them contains the name.
+By default the whole value of the attribute is expected to be the user name.
+If this is not the case, use the `format` option.
+It must contain one and only one pattern variable `%u` which will be replaced by the user name.
+
+### `auth.ldap.filter`
+* **Syntax:** string
+* **Default:** not set
+* **Example:** `filter = "(&(objectClass=shadowAccount)(memberOf=Jabber Users))"`
+
+An additional LDAP filter used to narrow down the search for user accounts.
+Do not forget to close the brackets and do not use superfluous whitespaces as this expression is processed before sending to LDAP - the match for user name (see `ldap.uids`) is added automatically.
+
+### `auth.ldap.dn_filter`
+* **Syntax:** TOML table with the following content:
+    * `filter` - string (LDAP filter), mandatory
+    * `attributes` - array of strings (attribute names)
+* **Default:** not set
+* **Example:** `dn_filter = {filter = "(&(name=%s)(owner=%D)(user=%u@%d))", attributes = ["sn"]}`
+
+This filter is applied to the results returned by the main filter.
+It performs an additional LDAP lookup to provide the complete result.
+This is useful when you are unable to define all filter rules in `ldap.filter`.
+You can define `%u`, `%d`, `%s` and `%D` pattern variables in the filter:
+
+* `%u` is replaced by the user’s part of a JID,
+* `%d` is replaced by the corresponding domain (virtual host),
+* `%s` variables are consecutively replaced by values of the attributes listen as `attributes`
+* `%D` is replaced by the Distinguished Name.
+
+Since this filter makes additional LDAP lookups, use it only as the last resort; try to define all filter rules in `ldap.filter` if possible.
+
+### `auth.ldap.local_filter`
+* **Syntax:** TOML table with the following content:
+    * `operation` - string, mandatory, `"equal"` or `"notequal"`
+    * `attribute` - string, mandatory, LDAP attribute
+    * `values` - array of strings (attribute values)
+* **Default:** not set
+* **Example:** `local_filter = {operation = "equal", attribute = "accountStatus", values = ["enabled"]}`
+
+If you can’t use the `ldap.filter` due to performance reasons (the LDAP server has many users registered), you can use this local filter.
+The local filter checks an attribute in MongooseIM, not in LDAP, so this limits the load on the LDAP directory.
+
+The example above shows a filter which matches accounts with the "enabled" status.
+Another example is shown below - it matches any account that is neither "disabled" nor "blacklisted".
+It also shows the usage of TOML dotted keys, which is recommended when the inline table grows too big.
+
+```toml
+   local_filter.operation = "not_equal"`
```

Comment:
```suggestion
   local_filter.operation = "not_equal"
```

Reply:
It's interesting (and a bit worrying), but I found a mistake in the very same line: should be `"notequal"`